contracts/rewards/RewardsManager.sol

```diff
@@ -355,6 +355,11 @@ contract RewardsManager is RewardsManagerV4Storage, GraphUpgradeable, IRewardsMa
      * @return Rewards amount for an allocation
```

Comment:
Doing staking() requires a CALL and a few SLOAD from the Controller, maybe we could store in memory before querying the state and using it then in `getAllocation`

Reply:
`IStaking staking = staking();`